diesel_derives/src/lib.rs

```diff
@@ -394,10 +394,11 @@ pub fn derive_query_id(input: TokenStream) -> TokenStream {
 ///
 /// This trait can only be derived for structs, not enums.
 ///
-/// **When this trait is derived, it will assume that the order of fields on your
-/// struct match the order of the fields in the query. This means that field
+/// **When this trait is derived, it will assume that __all fields on your struct__
+/// matches __all fields in the query__, including the order. This means that field
 /// order is significant if you are using `#[derive(Queryable)]`. Field name has
-/// no effect.**
+/// no effect. Think `Eq`, if you have `a, b, c` in your schema, you need to have
+/// `a, b, c` in the struct derived, not `a, b` or `a, c, b`.**
```

Comment:
If we write something like that in the documentation, I would like to see a concrete example here, not something generic like this. This is not really easy to understand for anyone else.

Reply:
Added